components/announcement_bar/overage_users_banner/index.tsx

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import React from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+import {useDispatch, useSelector} from 'react-redux';
+
+import {getCurrentUser, isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
+import {GlobalState} from 'types/store';
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+import AnnouncementBar from 'components/announcement_bar/default_announcement_bar';
+import {calculateOverageUserActivated} from 'utils/overage_team';
+import {isCurrentLicenseCloud} from 'mattermost-redux/selectors/entities/cloud';
+import {savePreferences} from 'mattermost-redux/actions/preferences';
+import {makeGetCategory} from 'mattermost-redux/selectors/entities/preferences';
+import {PreferenceType} from '@mattermost/types/preferences';
+import {LicenseLinks, StatTypes, Preferences, AnnouncementBarTypes} from 'utils/constants';
+
+import './overage_users_banner.scss';
+
+type AdminHasWatchItArgs = {
+    preferenceName: string;
+    overagePreferences: PreferenceType[];
+    isWarningBanner: boolean;
+}
+
+const adminHasWatchIt = ({preferenceName, overagePreferences, isWarningBanner}: AdminHasWatchItArgs): boolean => {
```

Comment:
nit: given that the admin has to explicitly dismiss it, should we rename this so that it's not confused that it's only when they see it once :

```suggestion
const adminHasDismissed = ({preferenceName, overagePreferences, isWarningBanner}: AdminHasWatchItArgs): boolean => {
```